engine/src/main/java/com/datatorrent/stram/client/StramAppLauncher.java

```diff
@@ -560,29 +560,36 @@ public URLClassLoader loadDependencies()
     return cl;
   }
 
-  private void setTokenRefreshKeytab(LogicalPlan dag, Configuration conf) throws IOException
+  private void setTokenRefreshCredentials(LogicalPlan dag, Configuration conf) throws IOException
   {
-    String keytabPath;
-    if ((keytabPath = conf.get(StramClientUtils.KEY_TAB_FILE)) == null) {
-      String keytab;
-      if ((keytab = StramUserLogin.getKeytab()) == null) {
+    String principal;
+    String keytab = null;
+    String keytabPath = conf.get(StramClientUtils.KEY_TAB_FILE);
+    if ((principal = StramUserLogin.getPrincipal()) == null) {
+      principal = conf.get(StramUserLogin.DT_AUTH_PRINCIPAL);
```

Comment:
Should the logic to get principal and keytab be encapsulated into StramUserLogin? For example:
```
  public static String getPrincipal(Configuration conf)
  {
    if (principal == null) {
      principal = conf.get(StramUserLogin.DT_AUTH_PRINCIPAL);
    }
    return principal;
  }

  public static String getKeytab(Configuration conf)
  {
    if (keytab == null) {
      keytab = conf.get(StramUserLogin.DT_AUTH_KEYTAB);
    }
    return keytab;
  }
```